server/src/test/java/org/opensearch/index/engine/InternalEngineTests.java

```diff
@@ -574,7 +573,7 @@ public void testMergeSegmentsOnCommit() throws Exception {
         final Settings.Builder settings = Settings.builder()
             .put(defaultSettings.getSettings())
             .put(IndexSettings.INDEX_MERGE_ON_FLUSH_MAX_FULL_FLUSH_MERGE_WAIT_TIME.getKey(), TimeValue.timeValueMillis(5000))
-            .put(IndexSettings.INDEX_MERGE_ON_FLUSH_ENABLED.getKey(), true);
+            .put(IndexSettings.INDEX_MERGE_ON_FLUSH_POLICY.getKey(), "merge-on-flush");
```

Comment:
Do we need to switch away from a boolean here at this point? I'm not sure we expect any merge on flush policy variations other than the default?

Reply:
@nknize the `INDEX_MERGE_ON_FLUSH_POLICY` is a new setting (semantics of `INDEX_MERGE_ON_FLUSH_ENABLED` is kept as boolean). The reason for that is that not every policy may implement merge-on-flush/commit but the composite one could be used for this reasons.